packages/scene-composer/src/components/modals/DeleteComponentModal.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useCallback, useContext } from 'react';
+import { useIntl } from 'react-intl';
+
+import { sceneComposerIdContext } from '../../common/sceneComposerIdContext';
+import { useStore } from '../../store';
+import { getLocalizedComponentType } from '../../common/componentTypeStings';
+
+import DeleteConfirmationModal from './DeleteConfirmationModal';
+
+const DeleteComponentModal: React.FC = () => {
+  const sceneComposerId = useContext(sceneComposerIdContext);
+  const intl = useIntl();
+  const getSceneNodeByRef = useStore(sceneComposerId)((state) => state.getSceneNodeByRef);
+  const setDeleteConfirmationModalVisible = useStore(sceneComposerId)(
+    (state) => state.setDeleteConfirmationModalVisible,
+  );
+  const deleteConfirmationModalParams = useStore(sceneComposerId)((state) => state.deleteConfirmationModalParams);
+  const removeComponent = useStore(sceneComposerId)((state) => state.removeComponent);
+
+  const nodeRef = deleteConfirmationModalParams?.nodeRef;
+  const node = getSceneNodeByRef(nodeRef);
+  const nodeName = node?.name ?? '';
+  const component = node?.components.find((c) => c.ref === deleteConfirmationModalParams?.componentRef);
+  const componentName = (component && getLocalizedComponentType(component, intl)) ?? '';
+
+  const title = intl.formatMessage(
+    {
+      defaultMessage: "Delete {nodeName}'s {componentName} component",
+      description: 'Delete component confirmation header text',
+    },
+    { nodeName, componentName },
+  );
+
+  const onDelete = useCallback(() => {
+    if (nodeRef && component?.ref) {
+      removeComponent(nodeRef, component.ref);
+    }
+    setDeleteConfirmationModalVisible(false);
+  }, [nodeRef, removeComponent, component?.ref]);
+
+  const contentMessage = useCallback(() => {
+    return intl.formatMessage(
+      {
+        defaultMessage:
+          "Delete {nodeName}'s {componentName} component? While this action may be undone, it may not return to its original state.",
+        description: 'Delete action confirmation message',
+      },
+      { componentName, nodeName },
+    );
+  }, [componentName, intl.formatMessage, nodeName]);
+
+  return (
+    <DeleteConfirmationModal
+      title={title}
+      contentBody={contentMessage()}
+      warningMessage={intl.formatMessage(
```

Comment:
NIT: This can also be moved out as a `const` same as `title` to maintain consistency.

Reply:
I'll move in next change